src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/workflow/GerritReviewStep.java

```diff
@@ -0,0 +1,144 @@
+ * THE SOFTWARE.
+ */
+package com.sonyericsson.hudson.plugins.gerrit.trigger.workflow;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.ToGerritRunListener;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Run;
+import javax.annotation.CheckForNull;
+import javax.inject.Inject;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Allows altering the Gerrit review posted at the end of build during the build.
+ * @author Teemu Murtola &lt;teemu.murtola@gmail.com&gt
+ */
+public class GerritReviewStep extends AbstractStepImpl {
+
+    private String customUrl;
+    private String unsuccessfulMessage;
+
+    /**
+     * Constructor.
+     *
+     * There are no mandatory parameters to the step.
+     */
+    @DataBoundConstructor
+    public GerritReviewStep() {
+    }
+
+    /**
+     * Gets the custom URL for a step.
+     * @return the URL.
+     */
+    @CheckForNull
+    public String getCustomUrl() {
+        return customUrl;
+    }
+
+    /**
+     * Sets a custom URL to post for a build.
+     * @param customUrl the URL to post.
+     */
+    @DataBoundSetter
+    public void setCustomUrl(String customUrl) {
+        this.customUrl = Util.fixEmptyAndTrim(customUrl);
+    }
+
+    /**
+     * Gets the unsuccessful message for a step.
+     * @return the message.
+     */
+    @CheckForNull
+    public String getUnsuccessfulMessage() {
+        return unsuccessfulMessage;
+    }
+
+    /**
+     * Sets additional information to post for a failed build.
+     * @param unsuccessfulMessage Additional information to post for a failed build.
+     */
+    @DataBoundSetter
+    public void setUnsuccessfulMessage(String unsuccessfulMessage) {
+        this.unsuccessfulMessage = Util.fixEmptyAndTrim(unsuccessfulMessage);
+    }
+
+    /**
+     * Executes the GerritReviewStep.
+     */
+    public static class Execution extends AbstractSynchronousStepExecution<Void> {
+
+        @StepContextParameter
+        private transient Run build;
+
+        @Inject
+        private transient GerritReviewStep step;
+
+        @Override
+        protected Void run() throws Exception {
+            ToGerritRunListener listener = ToGerritRunListener.getInstance();
+            String customUrl = step.getCustomUrl();
+            if (customUrl != null) {
+                listener.setBuildCustomUrl(build, customUrl);
+            }
+            String unsuccessfulMessage = step.getUnsuccessfulMessage();
+            if (unsuccessfulMessage != null) {
+                listener.setBuildFailureMessage(build, unsuccessfulMessage);
```

Comment:
Yes `obtainFailureMessage()` can be renamed without any problems, for the public methods we should make an effort to stay backwards compatible, e.g. keep the original method deprecated and just call the new method from it, just to be safe.

But I would rather that we are consistent and a bit wrong than inconsistent and a bit right :) Meaning you should only rename this new method if you feel you have the time to make all the corrections to stay consistent.
I would be very happy if you have time to make those corrections, but I don't expect you to do that work if you don't want to.
So only do the rename (of the new method) if you feel you have time/want to do the other rename work as well. Unfortunately I don't have much spare time to do it myself, so it's better to leave it consistent and a bit wrong in that case :) .